lib/containers/runtime.pm

```diff
@@ -0,0 +1,142 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Test Docker’s btrfs storage driver features for image and container management
+# Maintainer: qac team <qa-c@suse.de>
+
+package containers::runtime;
+use Mojo::Base -base;
+use testapi;
+use Test::Assert 'assert_equals';
+
+has runtime => undef;
+
+sub _rt_assert_script {
+    my ($self, $cmd, @args) = @_;
+    assert_script_run($self->runtime . " " . $cmd, @args);
+}
+
+sub _rt_script_run {
+    my ($self, $cmd, @args) = @_;
+    return script_run($self->runtime . " " . $cmd, @args);
+}
+
+sub _rt_script_output {
+    my ($self, $cmd, @args) = @_;
+    return script_output($self->runtime . " " . $cmd, @args);
+}
+
+=head2 build
+
+Build a container after visit C<dockerfile_path>.
+C<dockerfile_path> is the directory with the Dockerfile.
+C<container_tag> will be the name of the container.
+
+=cut
+sub build {
+    my ($self, $dockerfile_path, $container_tag) = @_;
+    die 'wrong number of arguments' if @_ < 3;
+    assert_script_run "cd $dockerfile_path";
```

Comment:
you entering directory in build function and looks like all other functions will depend on the fact that this one is called . I have doubts that it is always good idea . Can we have cases that we building without doing something else ? or we have some cases where we starting already existing container so we will skip build call ?

Reply:
I am not sure i follow. when you build a container you need to be into a directory that has a Dockerfile. The rest of the commands have nothing to do with the directory after that. You are free to use to perform any other action against any other image.

Reply:
so can you please leave directory than ? I think it would be cleaner

Reply:
maybe `pushd` and `popd` are could be your friends as we are going to perform a `cd` dance

Reply:
```
docker build [OPTIONS] PATH | URL
```

I guess it should work with the absolute path (let's be verbose) to Dockerfile.

Reply:
> ```
> docker build [OPTIONS] PATH | URL
> ```
> 
> I guess it should work with the absolute path (let's be verbose) to Dockerfile.

yeah I like this even more

Reply:
i change it but i havent covered everything. i am running a VR to see whether it breaks anything.